test/src/test/java/hudson/cli/ReloadJobCommandTest.java

```diff
@@ -0,0 +1,224 @@
+                .authorizedTo(Job.READ, Job.CONFIGURE, Jenkins.READ)
+                .invokeWithArgs("aProject1", "aProject2", "aProject3");
+
+        assertThat(result, succeededSilently());
+
+        assertThat(project1.scheduleBuild2(0).get().getLog(), containsString("echo 2"));
+        assertThat(project2.scheduleBuild2(0).get().getLog(), containsString("echo 2"));
+        assertThat(project3.scheduleBuild2(0).get().getLog(), containsString("echo 2"));
+    }
+
+    @Test public void reloadJobManyShouldFailIfAJobDoesNotExist() throws Exception {
+
+        FreeStyleProject project1 = j.createFreeStyleProject("aProject1");
+        project1.getBuildersList().add(new Shell("echo 1"));
+        FreeStyleProject project2 = j.createFreeStyleProject("aProject2");
+        project2.getBuildersList().add(new Shell("echo 1"));
+
+        assertThat(project1.scheduleBuild2(0).get().getLog(), containsString("echo 1"));
+        assertThat(project2.scheduleBuild2(0).get().getLog(), containsString("echo 1"));
+
+        changeProjectOnTheDisc(project1, "echo 1", "echo 2");
+        changeProjectOnTheDisc(project2, "echo 1", "echo 2");
+
+        final CLICommandInvoker.Result result = command
+                .authorizedTo(Job.READ, Job.CONFIGURE, Jenkins.READ)
+                .invokeWithArgs("aProject1", "aProject2", "never_created");
+
+        assertThat(result, failedWith(-1));
+        assertThat(result, hasNoStandardOutput());
+        assertThat(result.stderr(), containsString("No such job 'never_created'"));
+
+        assertThat(project1.scheduleBuild2(0).get().getLog(), containsString("echo 2"));
+        assertThat(project2.scheduleBuild2(0).get().getLog(), containsString("echo 2"));
+    }
+
+    @Test public void reloadJobManyShouldSucceedEvenAJobIsSpecifiedTwice() throws Exception {
+
+        FreeStyleProject project1 = j.createFreeStyleProject("aProject1");
+        project1.getBuildersList().add(new Shell("echo 1"));
+        FreeStyleProject project2 = j.createFreeStyleProject("aProject2");
+        project2.getBuildersList().add(new Shell("echo 1"));
+
+        assertThat(project1.scheduleBuild2(0).get().getLog(), containsString("echo 1"));
+        assertThat(project2.scheduleBuild2(0).get().getLog(), containsString("echo 1"));
+
+        changeProjectOnTheDisc(project1, "echo 1", "echo 2");
+        changeProjectOnTheDisc(project2, "echo 1", "echo 2");
+
+        final CLICommandInvoker.Result result = command
+                .authorizedTo(Job.READ, Job.CONFIGURE, Jenkins.READ)
+                .invokeWithArgs("aProject1", "aProject2", "aProject1");
+
+        assertThat(result, succeededSilently());
+
+        assertThat(project1.scheduleBuild2(0).get().getLog(), containsString("echo 2"));
+        assertThat(project2.scheduleBuild2(0).get().getLog(), containsString("echo 2"));
+    }
+
+    private void changeProjectOnTheDisc(final FreeStyleProject p, final String oldstr,
```

Comment:
Nit: the implementation can be simplified using `FilePath#readToString`/`FilePath#write(String, String)`.